zerver/lib/send_email.py

```diff
@@ -63,6 +63,10 @@ def build_email(template_prefix: str, to_user_ids: Optional[List[int]]=None,
         to_users = [get_user_profile_by_id(to_user_id) for to_user_id in to_user_ids]
         to_emails = [formataddr((to_user.full_name, to_user.delivery_email)) for to_user in to_users]
 
+    extra_headers = {}
+    if len(set([to_user.realm_id for to_user in to_users])) == 1:
+        extra_headers['X-Zulip-Organization'] = to_users[0].realm.string_id
```

Comment:
I don't like the idea of this auto-detection logic; do we have any emails that we build where `to_users` will be in multiple realms?  I think the answer should be no (since I believe we'll call this in a loop if we're e.g. sending an email to many users across multiple organizations); maybe we should just make it an assert?  

Or perhaps better, maybe we should make `realm` be an argument passed into these functions from the caller?  That would allow e.g. Zulip invitation emails to set this header (well, except for the rare corner case of realm-creation emails).

(Also, we should make at least one email test verify the `X-Zulip-Organization` header).